modules/xarray/fimex_xarray/xr.py

```diff
@@ -0,0 +1,171 @@
+    pyfimex0.CDMDataType.DOUBLE: np.float64,
+    pyfimex0.CDMDataType.INT: np.int,
+    pyfimex0.CDMDataType.INT64: np.int64,
+    pyfimex0.CDMDataType.UINT: np.uint,
+    pyfimex0.CDMDataType.UINT64: np.uint64,
+    pyfimex0.CDMDataType.SHORT: np.short,
+    pyfimex0.CDMDataType.USHORT: np.ushort,
+    pyfimex0.CDMDataType.STRING: str,
+    pyfimex0.CDMDataType.CHAR: np.byte,
+    pyfimex0.CDMDataType.UCHAR: np.ubyte,
+}
+
+
+class FimexBackendEntrypoint(BackendEntrypoint):
+    def open_dataset(self, filename_or_obj, drop_variables, *, config=None, filetype=None, decode_times=True, scale_offset=True):
+        if filetype is None:
+            filetype = ""
+        if config is None:
+            config = ""
+        fh = pyfimex0.createFileReader(filetype, filename_or_obj, config)
+        cdm = fh.getCDM()
+
+        var_names = set(cdm.getVariableNames())
+        attr_names = cdm.getGlobalAttributeNames()
+
+        attrs = dict()
+        for aname in attr_names:
+            attr = cdm.getGlobalAttribute(aname)
+            attrv = attr.getData().values()
+            attrs[aname] = attrv
+
+        coord_names = cdm.getDimensionNames()
+        coords = dict()
+        for cname in coord_names:
+            var = cdm.getVariable(cname)
+            values = var.getData().values()
+            shape = var.getShape()
+            if len(shape) != 1:
+                raise Exception()
+
+            if scale_offset:
+                try:
+                    scale_factor = cdm.getAttribute(cname, "scale_factor").getData().values()
+                except RuntimeError:
+                    scale_factor = 1
+                try:
+                    add_offset = cdm.getAttribute(cname, "add_offset").getData().values()
+                except RuntimeError:
+                    add_offset = 0
+                values = values*scale_factor + add_offset
+            coords[cname] = values
+            try:
+                var_names.remove(cname)
+            except KeyError:
+                pass
+
+        vars = dict()
+        for vname in var_names:
+            var = cdm.getVariable(vname)
+            dims = list(reversed(var.getShape()))
+            attrs = dict()
+            for aname in cdm.getAttributeNames(vname):
+                attr = cdm.getAttribute(vname, aname)
+                attrv = attr.getData().values()
+                if len(attrv) == 1:
+                    attrv = attrv[0]
+                attrs[aname] = attrv
+
+            data = FimexDataVariable(fh, cdm, var, scale_offset=scale_offset)
+            data = xarray.core.indexing.LazilyIndexedArray(data)
+            var = xarray.Variable(data=data, dims=dims, attrs=attrs)
+            vars[vname] = var
+
+        if decode_times:
+            if "time" in coord_names:
+                var = coords["time"]
+                var = np.array([np.datetime64(int(v), "s") for v in var])
+
+                coords["time"] = var
+
```

Comment:
I don't really know what this should do in xarray, but the time-variable in netcdf is every variable with a time unit, i.e. 'days since 2005-09-03 +15:00'. One cannot rely on the dimension-name, and that there is only one "time"-axis per file.
Fimex supports conversion of the time-units, i.e. by getScaledDataSizeInUnit(...) so you can get a unified handling (needs doubles or quads, I would say). Not really sure what you do her (expect time to be epoch-seconds?).